src/Microsoft.NetCore.Analyzers/Core/Security/UseSharedAccessProtocolHttpsOnly.cs

```diff
@@ -0,0 +1,140 @@
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.UseSharedAccessProtocolHttpsOnlyMessage),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.UseSharedAccessProtocolHttpsOnlyDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var microsoftWindowsAzureStorageNamespaceSymbol = compilationStartAnalysisContext
+                            .Compilation
+                            .GlobalNamespace
+                            .GetMembers("Microsoft")
+                            ?.FirstOrDefault()
+                            .GetMembers("WindowsAzure")
+                            .OfType<INamespaceOrTypeSymbol>()
+                            .FirstOrDefault()
+                            .GetMembers("Storage")
+                            .OfType<INamespaceOrTypeSymbol>()
+                            .FirstOrDefault();
+
+                if (microsoftWindowsAzureStorageNamespaceSymbol == null)
+                {
+                    return;
+                }
+
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+
+                compilationStartAnalysisContext.RegisterOperationBlockStartAction(operationBlockStartContext =>
+                {
+                    var owningSymbol = operationBlockStartContext.OwningSymbol;
+                    var interproceduralAnalysisConfig = InterproceduralAnalysisConfiguration.Create(
```

Comment:
Move this declaration close to the place where it is used. It is fine to create one instance per each DFA invocation, this is very lightweight type.